mods/default/functions.lua

```diff
@@ -210,6 +210,37 @@ end
 
 
 --
+-- fence registration helper
+--
+function default.register_fence(name, def)
+	local fence_texture = "default_fence_overlay.png^" .. def.texture .. "^default_fence_overlay.png^[makealpha:255,126,126"
+	minetest.register_node(name, {
+		description = def.description,
+		drawtype = "fencelike",
+		tiles = {def.texture},
+		inventory_image = fence_texture,
+		wield_image = fence_texture,
+		paramtype = "light",
+		sunlight_propagates = true,
+		is_ground_content = false,
+		selection_box = {
+			type = "fixed",
+			fixed = {-1/7, -1/2, -1/7, 1/7, 1/2, 1/7},
+		},
+		groups = def.groups,
+		sounds = def.sounds,
+	})
```

Comment:
Instead of just copying over certain spcific fields, how about just adding the needed fields to the definition?  This way you'll be able to do things like add glowglass fences.